pkg/build/api/types.go

```diff
@@ -111,6 +111,12 @@ type BuildSource struct {
 	// This allows to have buildable sources in directory other than root of
 	// repository.
 	ContextDir string `json:"contextDir,omitempty"`
+
+	// SourceSecretName is the name of a Secret that would be used for setting
+	// up the authentication for cloning private repository.
+	// The secret contains private ssh key, whereas the public key is associated
+	// with your SCM account.
+	SourceSecretName string `json:"sourceSecretName,omitempty"`
```

Comment:
But I'll prepare the mechanism for easily adding the rest of the methods, so in the end that description will be generally for all of the methods.